src/Nethermind/Nethermind.JsonRpc/JsonRpcService.cs

```diff
@@ -106,6 +108,75 @@ private Task<JsonRpcResponse> ExecuteRequestAsync(JsonRpcRequest rpcRequest, Jso
 
     private async Task<JsonRpcResponse> ExecuteAsync(JsonRpcRequest request, string methodName, ResolvedMethodInfo method, JsonRpcContext context)
     {
+        const string GetLogsMethodName = "eth_getLogs";
+
+        JsonRpcErrorResponse? value = PrepareParameters(request, methodName, method, out object[]? parameters, out bool hasMissing);
+        if (value is not null)
+        {
+            return value;
+        }
+
+        IRpcModule rpcModule = await _rpcModuleProvider.Rent(methodName, method.ReadOnly);
+        if (rpcModule is IContextAwareRpcModule contextAwareModule)
+        {
+            contextAwareModule.Context = context;
+        }
+        bool returnImmediately = methodName != GetLogsMethodName;
+        Action? returnAction = returnImmediately ? null : () => _rpcModuleProvider.Return(methodName, rpcModule);
+        IResultWrapper? resultWrapper = null;
+        try
+        {
+            // Execute method
+            object invocationResult = hasMissing ?
+                method.MethodInfo.Invoke(rpcModule, parameters) :
+                method.Invoker.Invoke(rpcModule, new Span<object?>(parameters));
```

Comment:
Can we make it faster in any way? Compile to delegates?

Reply:
Will look into it the `method.Invoker` path is pretty fast (MethodInfo one is slow)
<img width="607" height="297" alt="image" src="https://github.com/user-attachments/assets/e304d018-f6cf-4f38-9f60-21bb5ff1865e" />